plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java

```diff
@@ -171,7 +172,8 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLRegi
                 "To upload custom model user needs to enable allow_registering_model_via_url settings. Otherwise please use OpenSearch pre-trained models."
             );
         }
-        registerModelInput.setIsHidden(RestActionUtils.isSuperAdminUser(clusterService, client));
+        boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
+        registerModelInput.setIsHidden(isSuperAdmin);
```

Comment:
Should we check hidden field in model configuration? 
Is it possible super admin create non-hidden model?

Reply:
In current design this is not possible for super admin to create non-hidden model.